File: Csharp/HealthTech/Usuario.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HealtTech
{
    using System;
    using System.Collections.Generic;
    using System.IO;
    using CsvHelper;
    using CsvHelper.Configuration;
    using System.Globalization;
    using System.Linq;

    public class USUARIO
    {
        public string Usuario { get; set; }
        public string Contrasena { get; set; }
        private string rutaUsuarios = @"C:\Users\Luis Angel Seoanes O\OneDrive\Escritorio\HealtTech\Csharp\Archivos\Usuarios.csv"; // Ruta del archivo CSV

        public USUARIO(string usuario, string contrasena)
        {
            Usuario = usuario;
            Contrasena = contrasena;
        }

        // Cargar usuarios desde el archivo CSV
        public Dictionary<string, Dictionary<string, object>> CargarUsuarios()
        {
            var usuarios = new Dictionary<string, Dictionary<string, object>>();
            if (File.Exists(rutaUsuarios))
            {
                using (var reader = new StreamReader(rutaUsuarios))
                using (var csv = new CsvReader(reader, new CsvConfiguration(CultureInfo.InvariantCulture)))
                {
                    csv.Read();
                    csv.ReadHeader();

                    while (csv.Read())
                    {
                        var usuarioData = new Dictionary<string, object>
                    {
                        { "ID", csv.GetField<int>("ID") },
                        { "Contrasena", csv.GetField<string>("Contrasena") },
                        { "Tcuenta", csv.GetField<string>("Tcuenta") }
                    };

                        usuarios[csv.GetField<string>("Usuario")] = usuarioData;
                    }
                }
            }
            return usuarios;
        }

        // Verificar inicio de sesión
        public bool InicioSesion()
        {
            var usuarios = CargarUsuarios();
            if (usuarios.ContainsKey(Usuario))
            {
                if (usuarios[Usuario]["Contrasena"].ToString() == Contrasena)
                {
                    Console.WriteLine("Inicio de sesión exitoso.");
                    return true;
                }
                else
                {
                    Console.WriteLine("Contraseña incorrecta.");
                    return false;
                }
            }
            else
            {
                Console.WriteLine("El usuario no existe.");
                return false;
            }
        }

        // Guardar los usuarios en el archivo CSV
        public void GuardarUsuarios(Dictionary<string, Dictionary<string, object>> usuarios)
        {
            using (var writer = new StreamWriter(rutaUsuarios))
            using (var csv = new CsvWriter(writer, new CsvConfiguration(CultureInfo.InvariantCulture)))
            {
                csv.WriteField("ID");
                csv.WriteField("Usuario");
                csv.WriteField("Contrasena");
                csv.WriteField("Tcuenta");
                csv.NextRecord();

                foreach (var usuario in usuarios)
                {
                    csv.WriteField(usuario.Value["ID"]);
                    csv.WriteField(usuario.Key);
                    csv.WriteField(usuario.Value["Contrasena"]);
                    csv.WriteField(usuario.Value["Tcuenta"]);
                    csv.NextRecord();
                }
            }
        }

        // Crear una nueva cuenta de usuario
        public bool CrearCuenta(string tipoCuenta)
        {
            var usuarios = CargarUsuarios();
            if (usuarios.ContainsKey(Usuario))
            {
                Console.WriteLine("El usuario ya existe.");
                return false;
            }
            else
            {
                // Determinar el nuevo ID basado en el ID más alto actual
                int nuevoId = usuarios.Count == 0 ? 1 : usuarios.Values.Max(u => (int)u["ID"]) + 1;

                // Agregar el nuevo usuario
                usuarios[Usuario] = new Dictionary<string, object>
            {
                { "ID", nuevoId },
                { "Contrasena", Contrasena },
                { "Tcuenta", tipoCuenta }
            };

                // Guardar los usuarios actualizados
                GuardarUsuarios(usuarios);

                Console.WriteLine($"Usuario {Usuario} creado con éxito con ID {nuevoId}.");
                return true;
            }
        }
    }

}

File: Usuario.py
import csv
import os

class USUARIO:
    def __init__(self, usuario, contrasena):
        self.usuario = usuario
        self.contrasena = contrasena
        self.ruta_usuarios = 'Archivos/Usuarios.csv'  # Archivo CSV para almacenar usuarios

    def cargarUsuarios(self):
        """Cargar todos los usuarios desde el archivo CSV."""
        usuarios = {}
        if os.path.exists(self.ruta_usuarios):
            with open(self.ruta_usuarios, mode='r', newline='', encoding='utf-8') as file:
                reader = csv.DictReader(file)
                for row in reader:
                    usuarios[row['Usuario']] = {
                        "ID": int(row['ID']),  # Convertimos el ID a entero
                        "Contrasena": row['Contrasena'],
                        "Tcuenta": row['Tcuenta']
                    }
        return usuarios

    def InicioSesion(self):
        """Verificar si el usuario existe y si la contraseña es correcta."""
        usuarios = self.cargarUsuarios()
        if self.usuario in usuarios:
            if usuarios[self.usuario]["Contrasena"] == self.contrasena:
                print("Inicio de Sesión exitoso.")
                return True
            else:
                print("Contraseña incorrecta.")
                return False
        else:
            print("El usuario no existe.")
            return False


            

    def guardarUsuarios(self, usuarios):
        """Guardar todos los usuarios en el archivo CSV."""
        with open(self.ruta_usuarios, mode='w', newline='', encoding='utf-8') as file:
            fieldnames = ['ID', 'Usuario', 'Contrasena', 'Tcuenta']  # Aseguramos que los nombres de campo coincidan
            writer = csv.DictWriter(file, fieldnames=fieldnames)
            writer.writeheader()
            for usuario, datos in usuarios.items():
                writer.writerow({
                    'ID': datos['ID'],
                    'Usuario': usuario,
                    'Contrasena': datos['Contrasena'],
                    'Tcuenta': datos['Tcuenta'],
                })

    def CrearCuenta(self, TipoCuenta):
        """Crear una nueva cuenta y guardarla en el archivo CSV."""
        usuarios = self.cargarUsuarios()
        if self.usuario in usuarios:
            print("El usuario ya existe.")
            return False
        else:
            # Determinamos el nuevo ID basado en el ID más alto actual
            if not usuarios:
                Nuevoid = 1  # Si no hay usuarios, empezamos con el ID 1
            else:
                max_id = max(datos["ID"] for datos in usuarios.values())
                Nuevoid = max_id + 1

            # Agregamos el nuevo usuario
            usuarios[self.usuario] = {
                "ID": Nuevoid,
                "Contrasena": self.contrasena,
                "Tcuenta": TipoCuenta,
            }
            # Guardamos la lista de usuarios actualizada
            self.guardarUsuarios(usuarios)
            print(f"Usuario {self.usuario} creado con éxito con ID {Nuevoid}.")
            return True
